arc4.c

```diff
@@ -128,40 +128,6 @@ arc4_ARC4_dealloc(struct arc4_ARC4 *self)
     Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
-static inline int
-byteslike_as_string_and_size(PyObject *obj, char **buffer, Py_ssize_t *size)
-{
-#if PY_MAJOR_VERSION >= 3
-    const char *utf8;
-#else
-    PyObject *ascii;
-#endif
-
-    if (PyBytes_Check(obj)) {
-        *buffer = PyBytes_AS_STRING(obj);
-        *size = PyBytes_GET_SIZE(obj);
-        return 0;
-    }
-    if (PyUnicode_Check(obj)) {
-#if PY_MAJOR_VERSION >= 3
-        utf8 = (char *)PyUnicode_AsUTF8AndSize(obj, size);
-        if (utf8 != NULL) {
-            *buffer = (char *)utf8;
-            return 0;
-        }
-#else
-        ascii = PyUnicode_AsASCIIString(obj);
-        if (ascii != NULL) {
-            *buffer = PyBytes_AS_STRING(ascii);
-            *size = PyBytes_GET_SIZE(ascii);
-            Py_DECREF(ascii);
-            return 0;
-        }
```

Comment:
1. `ascii` was created with refcount 1. (L153)
2. `buffer` points at the internal buffer of `ascii`. (L155)
3. `Py_DECREF(ascii)` decrements `ascii`'s refcount to 0. (L157)
4. `ascii` is released and its internal buffer gets uninitialized.